File: src/main/java/org/droidmate/analyzer/evaluation/IEvaluationStrategy.java
package org.droidmate.analyzer.evaluation;

import org.droidmate.analyzer.exploration.IExplorationResult;


/**
 * Interface defining metrics to evaluate scenarios. This evaluation is used to define which
 * scenarios can be combined on @org.droidmate.analyzer.exploration.DefaultExplorationStrategy.generateScenarios
 */
public interface IEvaluationStrategy {
    /**
     * Check if a scenario output is valid
     * @param result An exploration result
     * @return If the scenario is valid
     */
    boolean isValid(IExplorationResult result);

    double getDissimilarity(IExplorationResult result);
}

File: src/main/kotlin/org/droidmate/analyzer/evaluation/IEvaluationStrategy.kt
package org.droidmate.analyzer.evaluation

import org.droidmate.analyzer.exploration.IExplorationResult


/**
 * Interface defining metrics to evaluate scenarios. This evaluation is used to define which
 * scenarios can be combined on @org.droidmate.analyzer.exploration.DefaultExplorationStrategy.generateScenarios
 */
interface IEvaluationStrategy {
    /**
     * Check if a scenario output is valid
     * @param result An exploration result
     * *
     * @return If the scenario is valid
     */
    fun isValid(result: IExplorationResult): Boolean

    fun getDissimilarity(result: IExplorationResult): Double
}
